src/main.lua

```diff
@@ -420,6 +420,11 @@ end
 
 -- Sanitize inputs before every interaction
 local function assertAndSanitizeInputs(msg)
+	if msg.Tags.Action ~= "Eval" and msg.Data then
+		-- assert data size is less than or equal to 100 bytes, we dont use msg.Data
+		assert(#msg.Data <= 100, "Data size is too large")
```

Comment:
if the content type is JSON and a json string is provided, this could fail. Recommend adding a test specifically that large tx's with a json content type fail

Reply:
updated! nice catch

Reply:
@atticusofsparta why would this fail if msg.Data is a stringiifed json? we want to avoid encoding if we don't need to

Reply:
42b1432

Reply:
#msg.Data on lua type of table was indeed not asserting correctly when Content-Type application/json is fed through AOS

Reply:
@dtfiedler in the top level global handle, if the content type is set to application/json, it will decode the msg.Data field and overwrite it with a lua table.

Reply:
Now, it wouldn't actually break, but it would be inaccurate. For example if it was a table with 1 element, it would have a length of 1, but not a byte length of 1